Models/Certificate.swift

```diff
@@ -142,31 +142,43 @@ private enum MethodsForV1_1 {
     }
 
     static func parse(recipientJSON: AnyObject?) -> Recipient? {
+
         guard let recipientData = recipientJSON as? [String : AnyObject],
             let identityType = recipientData["type"] as? String,
             let familyName = recipientData["familyName"] as? String,
             let givenName = recipientData["givenName"] as? String,
-            let isHashed = recipientData["hashed"] as? String,
+            let isHashedObj = recipientData["hashed"] as AnyObject?,
             let publicKey = recipientData["pubkey"] as? String,
             let identity = recipientData["identity"] as? String else {
                 return nil
         }
         
+        var hashed : Bool = false
+
+        if let isHashed = isHashedObj as? Bool {
+            hashed = isHashed
+        } else {
+            hashed = isHashedObj.uppercased == "TRUE"
```

Comment:
I'm surprised the compiler didn't complain. I'm positive `isHashedObj` is of type `AnyObject` here, and I don't believe there's an instance method of `uppercased` on `AnyObject`, just on `String`.

It might be better to do a switch on the type of `isHashedObj`:
```

switch isHashedObj {
case let value as String:
    print("String")
case let value as Bool:
    print("Bool")
default:
    print("Uh oh")
}
```